### PR TITLE
feat: add possibility to route requests to sidecar container

### PR DESCRIPTION
**Pull request by edorizzardi**
# Description

This PR changes the `Service` such that it allows to route requests to an arbitrary port. This is not a breaking change.

### CODE REVIEW

**Comment by dmotte** [comment]
@pierluigilenoci since you are the most active committer of this repo, may we kindly ask you if you could please take a look? Although quite simple, this fix would be very beneficial for us 🙂 thank you so much in advance!

**Comment by pierluigilenoci** [comment]
@edorizzardi Please update the [message for ArtifactHub](https://github.com/oauth2-proxy/manifests/blob/595ee16ab9d68092bb447e4dc08764223ca6bfb7/helm/oauth2-proxy/Chart.yaml#L38) and rebase the PR from the main branch. 🙏🏻 👍🏻

**Comment by edorizzardi** [comment]
@pierluigilenoci Updated minor version. Plus I added the parameter in the list, and added a small paragraph on how to use my changes.

File: helm/oauth2-proxy/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: oauth2-proxy
-version: 7.8.3
+version: 7.9.3
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/Chart.yaml:2]: Please `7.9.0` 🙏🏻

File: helm/oauth2-proxy/README.md [hunk 2]
```diff
@@ -196,6 +196,7 @@ The following table lists the configurable parameters of the oauth2-proxy chart
...
 | `service.loadBalancerSourceRanges`                    | allowed source ranges in load balancer                                                                                                                                                                                                                           | `nil`                                                                            |
 | `service.nodePort`                                    | external port number for the service when service.type is `NodePort`                                                                                                                                                                                             | `nil`                                                                            |
+| `service.targetPort`                                  | name for the service's `targetPort`                                                                                                                                                                                                                              | `""`
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/README.md:199]: "A numeric port number (e.g., `80`) or a port name defined in the pod's container(s) (e.g., `http`)."

File: helm/oauth2-proxy/README.md [hunk 3]
```diff
@@ -357,3 +358,6 @@ config:
     whitelist_domains = [ ".domain.com", ".example.io"]
     provider = "google"
 ```
+
+## Route requests to sidecar container
+You can route requests to a sidecar container first by setting the `service.targetPort` variable; by default, the service's `targetPort` value equals to the `httpSchema`'s.
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/README.md:363]: The possible values for the `targetPort` field of a Kubernetes Service can be either a port number or the name of a port defined in the pod.

File: helm/oauth2-proxy/values.yaml [hunk 4]
```diff
@@ -153,6 +153,9 @@ service:
   externalTrafficPolicy: ""
   # configure internalTrafficPolicy
   internalTrafficPolicy: ""
+  # configure service target port
+  # useful when there's need to route requests to sidecar containers first
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/values.yaml:157]: I would remove this comment.

**Review by pierluigilenoci — COMMENTED** [review]
There are still some details to iron out, sorry if I'm being pedantic. 🙏🏻

File: helm/oauth2-proxy/Chart.yaml (hunk 1 above)
Comment (edorizzardi) [reply helm/oauth2-proxy/Chart.yaml]: fixed in [ca8385f](https://github.com/oauth2-proxy/manifests/pull/270/commits/ca8385f9dac9f7443d1b7176604bc7aec1c2bf72)

File: helm/oauth2-proxy/README.md (hunk 2 above)
Comment (edorizzardi) [reply helm/oauth2-proxy/README.md]: changed in [a423d1d](https://github.com/oauth2-proxy/manifests/pull/270/commits/a423d1d151aa387935ee105b7845f81d47d9564f). Let me know if it's good to you

File: helm/oauth2-proxy/README.md (hunk 3 above)
Comment (edorizzardi) [reply helm/oauth2-proxy/README.md]: Changed in [a423d1d](https://github.com/oauth2-proxy/manifests/pull/270/commits/a423d1d151aa387935ee105b7845f81d47d9564f)

File: helm/oauth2-proxy/values.yaml (hunk 4 above)
Comment (edorizzardi) [reply helm/oauth2-proxy/values.yaml]: fixed in [01147b8](https://github.com/oauth2-proxy/manifests/pull/270/commits/01147b849ac7afbc246a5d20b291a2ff2f8d8134)

**Review by pierluigilenoci — APPROVED** [review]

**Pull request merged**